### ISSUE TITLE
Automatically setting persistentConnection to false if request size > 64KiB

### ISSUE THREAD

**Issue by davidmigloz** [Zekfad/fetch_client#14]
Hello!

Thanks for the package, it's really useful.

I've wasted a couple of hours debugging failing requests and they ended up being caused by the fact that the request size was > 64KiB. I did read your 'Large payload' section in the README but it took me a while to realize that that was the problem as the error message was not helpful at all.

I was wondering what would be the downside of automatically setting `persistentConnection` to false if the request size is > 64KiB?

Something like:

```dart
// If the request if bigger than 60KiB set persistentConnection to false
if ((request.contentLength ?? 0) > 61440) {
  request.persistentConnection = false;
}
```

What do you think?

**Comment by Zekfad** [comment]
Hello, that is a good suggestion, but it isn't so easy to check for size.

`http` exposes user payload as [`ByteStream`](https://pub.dev/documentation/http/latest/http/ByteStream-class.html), and we cannot access length without fully reading the stream, which defeats the purpose - streaming of request.

It may be possible to use [`BaseRequest#contentLength`](https://pub.dev/documentation/http/latest/http/BaseRequest/contentLength.html), but it is also not always available, and have a following issue:
> This operation listens to this stream, and a non-broadcast stream cannot be reused after finding its length.

Using something that works only sometimes might lead to even more frustrating debugging, when one time your data sends ok, the other it's just fails with no good reason.

As for the error message I found it bad as well, but it's the one from the browser and there is not so much we can do, hence than message in the README.

**Comment by Zekfad** [comment]
As a possible workaround if `streamRequests` is disabled we can read payload and use assert to trigger error in debug mode if necessary, what do you think?

**Comment by davidmigloz** [comment]
Thanks for your reponse @Zekfad!

So we have the following cases:

| Request \ Response    | Non-streaming response | Streaming response |
|-----------------------|------------------------|--------------------|
| Non-streaming request | ✅                      | ✅                  |
| Streaming request     | ❌                      | ❌                  |

For the ones with ✅ (which I guess are also the most frequent), we should be able to get the `contentLength` without issues, shouldn't we?

The package could handle it automatically for those cases, so the developer would only care about it when streaming requests.

**Comment by Zekfad** [comment]
Will be fixed with 1.2.0 publish.

**Comment by davidmigloz** [comment]
I've tested it and works fine. Thank you!
